metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -146,48 +151,73 @@ def build_plan(
 
         return DataflowPlan(plan_id=plan_id, sink_output_nodes=[sink_node])
 
-    def _build_metrics_output_node(self, query_spec: MetricFlowQuerySpec) -> ComputedMetricsOutput[SqlDataSetT]:
-        """Build a node that computes all requested metrics along with linkables."""
-
+    def _build_metrics_output_node(
+        self,
+        metric_specs: Sequence[MetricSpec],
+        queried_linkable_specs: LinkableSpecSet,
+        where_constraint: Optional[SpecWhereClauseConstraint] = None,
+        time_range_constraint: Optional[TimeRangeConstraint] = None,
+    ) -> ComputedMetricsOutput[SqlDataSetT]:
         compute_metrics_nodes: List[ComputedMetricsOutput[SqlDataSetT]] = []
-        for metric_spec in query_spec.metric_specs:
+        for metric_spec in metric_specs:
             logger.info(f"Generating compute metrics node for {metric_spec}")
-            metric = self._metric_semantics.get_metric(metric_spec)
-            metric_input_measure_specs = self._metric_semantics.measures_for_metric(metric_spec)
-
-            logger.info(
-                f"For {metric_spec}, needed measures are:\n"
-                f"{pformat_big_objects(metric_input_measure_specs=metric_input_measure_specs)}"
-            )
+            # TODOmetricsem: refactor MetricSemantics to use MetricReference instead of MetricSpec as an accessor key
+            blank_metric_spec = MetricSpec.from_element_name(metric_spec.element_name)
+            metric = self._metric_semantics.get_metric(blank_metric_spec)
+            if metric.type == MetricType.DERIVED:
+                metric_input_specs = self._metric_semantics.metric_input_specs_for_metric(blank_metric_spec)
+                logger.info(
+                    f"For derived metric: {metric_spec}, needed metrics are:\n"
+                    f"{pformat_big_objects(metric_input_specs=metric_input_specs)}"
+                )
 
-            combined_where = query_spec.where_constraint
-            if metric.constraint:
-                metric_constraint = WhereConstraintConverter.convert_to_spec_where_constraint(
-                    self._data_source_semantics, metric.constraint
+                compute_metric_node = self._build_metrics_output_node(
```

Comment:
Ok, so this includes all metrics in the dataflow plan, but when we render the SQL we strip them out of the result sets for each metric node (i.e., every visit to the metric node returns only the results of that metric node, not any of its inputs).

Could we get some plan file test cases for this logic just so we can see how it all works?

Reply:
added plan files!